ingenialink/eoe/network.py

```diff
@@ -385,3 +388,25 @@ def load_firmware_moco(self) -> None:  # type: ignore [override]
 
     def load_firmware(self) -> None:  # type: ignore [override]
         raise NotImplementedError
+
+    def _recover_from_power_cycle(self, status: NET_DEV_EVT) -> None:
+        """Recover the connection after a power cycle.
+
+        Args:
+            status: The network status.
+
+        """
+        if status == NET_DEV_EVT.REMOVED:
+            connection_recovery_thread = Thread(target=self._connection_recovery)
+            connection_recovery_thread.start()
+
+    def _connection_recovery(self) -> None:
+        """Restart the EoE service until all slaves are detected."""
+        while not all(servo.is_alive() for servo in self.servos):
+            with contextlib.suppress(ILError):
+                self._stop_eoe_service()
+                self._deinitialize_eoe_service()
+                self._eoe_service_init = False
+                self._initialize_eoe_service()
+                self.__reconfigure_drives()
+                self._start_eoe_service()
```

Comment:
This happens for ever until all drives are recovered, right? Then, if any drive is disconnected for ever, EoE service is unusable until disconnect this drive, right?